src/mpyl/steps/deploy/k8s/cluster.py

```diff
@@ -0,0 +1,81 @@
+""" Utilities for creating rancher compatible helm charts. """
+
+from dataclasses import dataclass
+from typing import Optional, Any
+
+from ...models import RunProperties
+from ....project import TargetProperty, Project
+
+
+@dataclass(frozen=True)
+class ClusterConfig:
+    name: str
+    cluster_id: Optional[str]
+    cluster_env: str
+    context: str
+
+    @staticmethod
+    def from_config(config: dict):
+        return ClusterConfig(
+            name=config["name"],
+            cluster_id=config.get("clusterId"),
+            cluster_env=config["clusterEnv"],
+            context=config["context"],
+        )
+
+
+def get_cluster_config_for_project(
+    run_properties: RunProperties, project: Project
+) -> ClusterConfig:
+    cluster_override = (
+        project.deployment.cluster.get_value(run_properties.target)
+        if project.deployment and project.deployment.cluster
+        else None
+    )
+
+    kubernetes_config = run_properties.config["kubernetes"]
+
+    clusters = [
+        ClusterConfig.from_config(cluster_config)
+        for cluster_config in kubernetes_config["clusters"]
+    ]
+
+    default_cluster_name: str = TargetProperty.from_config(
+        kubernetes_config["defaultCluster"]
+    ).get_value(run_properties.target)
+
+    default_cluster: ClusterConfig = next(
+        cluster for cluster in clusters if cluster.name == default_cluster_name
+    )
+
+    if not default_cluster:
+        raise ValueError(
+            f"Default cluster {default_cluster_name} not found in list of clusters"
+        )
+
+    cluster_for_env = next(
+        (cluster for cluster in clusters if cluster.name == cluster_override),
+        default_cluster,
```

Comment:
Shouldn't we throw an error instead of using the default if an override is defined but it doesn't exist?

Reply:
hm I guess we could, but how do you differentiate a cluster override not being found vs not being defined?